galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/CPSRoute.java

```diff
@@ -184,10 +114,8 @@ protected String buildResponseBody(String namespace, Map<String, String> propert
          */
         JsonArray propertyArray = new JsonArray();
         for (Map.Entry<String, String> entry : properties.entrySet()) {
-            JsonObject cpsProp = new JsonObject();
-            cpsProp.addProperty("name", entry.getKey());
-            cpsProp.addProperty("value", getProtectedValue(entry.getValue(),namespace));
-            propertyArray.add(cpsProp);
+            GalasaProperty property = new GalasaProperty(entry.getKey(),propertyUtility.getProtectedValue(entry.getValue(),namespace));
+            propertyArray.add(property.toJSON());
```

Comment:
I don't think need the `property.toJSON()` since you could create a `List<GalasaProperty>` and plug that into gson directly like `gson.toJson(galasaProperties)`, and that will create the JSON array that you expect